File: contest-management-client-server-java/ContestMgmtModel/src/main/java/contestmgmt/model/Registration.java
package contestmgmt.model;

public class Registration extends Entity<Tuple<Long, Long>> {
    private Participant participant;
    private Competition competition;

    public Registration(Participant participant, Competition competition) {
        setId(new Tuple<>(participant.getId(), competition.getId()));
        this.participant = participant;
        this.competition = competition;
    }

    public Participant getParticipant() {
        return participant;
    }

    public void setParticipant(Participant participant) {
        this.participant = participant;
    }

    public Competition getCompetition() {
        return competition;
    }

    public void setCompetition(Competition competition) {
        this.competition = competition;
    }
}

File: contest-management-client-server-cs/model/Registration.cs
using System;

namespace model
{
    [Serializable]
    public class Registration : Entity<(long, long)>
    {
        private Participant _participant;

        private Competition _competition;

        public Registration(Participant participant, Competition competition)
        {
            Id = (participant.Id, competition.Id);
            _participant = participant;
            _competition = competition;
        }

        public Competition Competition
        {
            get => _competition;
            set
            {
                Id = (_participant.Id, value.Id);
                _competition = value;
            }
        }
        
        public Participant Participant
        {
            get => _participant;
            set
            {
                Id = (value.Id, _competition.Id);
                _participant = value;
            }
        }

        public void Deconstruct(out Participant participant, out Competition competition)
        {
            (participant, competition) = (Participant, Competition);
        }
    }
}